ehsm_kms_service/delete_secret_thread.js

```diff
@@ -0,0 +1,118 @@
+const {
+    base64_decode
+} = require('./function')
+
+// Set a queue to record the waiting time
+var delete_task_queue = {}
+
+/**
+ * remove a timer from exist timer by key
+ * @param {Object} DB : database control
+ * @param {Object} key : key form exist timer
+ * @returns
+ */
+const remove_delete_task = (secretName, appid) => {
+    try {
+        //Delete thread when thread exists
+        if (delete_task_queue[secretName + '_' + appid] != undefined) {
+            clearTimeout(delete_task_queue[secretName + '_' + appid].timer)
+        }
+        delete delete_task_queue[secretName + '_' + appid];
+    } catch (error) {
+        console.error('error remove_delete_task :: ', error)
+    }
+}
+
+/**
+ * public interface, add a new delete task into the delete queue with the secret name.
+ * @param {String} appid : appid of user
+ * @param {Object} DB : database control
+ * @param {Object} secretName : The name of the secret. eg. 'secretName01'
+ * @param {Object} plannedDeleteTime : planned delete time for secret eg. '1659931644416'
+ * @returns
+ */
+const add_delete_task = (DB, appid, secretName, plannedDeleteTime) => {
+    try {
+        let delayMillisecond = plannedDeleteTime - new Date().getTime()
+        let needAdd = true
+        let key = secretName + '_' + appid
+        // process exist timer, 
+        // If the new time is within plus or minus 10 seconds of the original time, keep the original plan unchanged; otherwise, clear the original plan and add again
+        if (delete_task_queue.hasOwnProperty(key)) {
```

Comment:
If a user pushes the delete time 8 seconds later and wants to restore the secret in the last 5 seconds. This would cause this secret cannot be found. Is this acceptable?

Reply:
Scheduled deletion time is at least 1 day，8 seconds is too short

Reply:
I schedule to delete recoveryPeriod=2 days on 2022.1.1 13:00:00
then I schedule to delete recoveryPeriod=1 on 2022.1.2 13:00:08
that will get a 8 seconds later.